app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
+
+  def my_any?
+    each do |j|
+      return true if yield(j)
+    end
+    false
+  end
+
+  def my_none?
+    condition = true
+    each do |j|
+      condition = false if yield(j)
+    end
+    condition
+  end
+
+  def my_count(counted)
+    counting = 0
+    each do |j|
+      counting += 1 if j == counted
+    end
+    counting
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    each do |i|
+      if !proc.nil?
+        arr.push(proc.call(i))
+      else
+        arr.push(yield i)
+      end
+    end
+    arr
+  end
```

Comment:
1.Should return an Enumerator if no block is given. For example:
 `p [1, 2, 3].my_map` should return `#<Enumerator: [1, 2, 3]:my_map>`